From 27e0c547c220f41e5be9f24ec33e089573c5b98f Mon Sep 17 00:00:00 2001
From: Rocky Hao <rocky.hao@rock-chips.com>
Date: Wed, 18 Apr 2018 16:27:23 +0800
Subject: [PATCH] arm64: dts: rockchip: update pinctrl for px30

1 move pinctrl into board level dts file.
2 remove pinctrl for sleep state.
in sleep state we do not change pin control and keep the pin control
in otp state, which is used by atf as a flag to control pmic's state.

Change-Id: Ib68b20d4f4ba79d99255f1deb509ff8a741deef2
Signed-off-by: Rocky Hao <rocky.hao@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/px30-evb-ddr3-lvds-v10.dts   | 3 +++
 arch/arm64/boot/dts/rockchip/px30-evb-ddr3-v10.dts        | 3 +++
 arch/arm64/boot/dts/rockchip/px30-evb-ddr4-v10.dts        | 3 +++
 arch/arm64/boot/dts/rockchip/px30.dtsi                    | 4 ----
 arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10.dts       | 3 +++
 arch/arm64/boot/dts/rockchip/rk3326-evb-ai-va-v10.dts     | 3 +++
 arch/arm64/boot/dts/rockchip/rk3326-evb-linux-lp3-v10.dts | 3 +++
 arch/arm64/boot/dts/rockchip/rk3326-evb-lp3-v10.dts       | 3 +++
 8 files changed, 21 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-lvds-v10.dts b/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-lvds-v10.dts
index 8c10e393bab3..b1485d5aba09 100644
--- a/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-lvds-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-lvds-v10.dts
@@ -577,6 +577,9 @@
 };
 
 &tsadc {
+	pinctrl-names = "init", "default";
+	pinctrl-0 = <&tsadc_otp_gpio>;
+	pinctrl-1 = <&tsadc_otp_out>;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-v10.dts b/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-v10.dts
index ca19d98fb401..da9bb00f2d7a 100644
--- a/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-v10.dts
@@ -807,6 +807,9 @@
 };
 
 &tsadc {
+	pinctrl-names = "init", "default";
+	pinctrl-0 = <&tsadc_otp_gpio>;
+	pinctrl-1 = <&tsadc_otp_out>;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/rockchip/px30-evb-ddr4-v10.dts b/arch/arm64/boot/dts/rockchip/px30-evb-ddr4-v10.dts
index d7676486a101..6c4376a43307 100644
--- a/arch/arm64/boot/dts/rockchip/px30-evb-ddr4-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/px30-evb-ddr4-v10.dts
@@ -730,6 +730,9 @@
 };
 
 &tsadc {
+	pinctrl-names = "init", "default";
+	pinctrl-0 = <&tsadc_otp_gpio>;
+	pinctrl-1 = <&tsadc_otp_out>;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/rockchip/px30.dtsi b/arch/arm64/boot/dts/rockchip/px30.dtsi
index 61f2110a9b9d..86effa681953 100644
--- a/arch/arm64/boot/dts/rockchip/px30.dtsi
+++ b/arch/arm64/boot/dts/rockchip/px30.dtsi
@@ -854,10 +854,6 @@
 		assigned-clock-rates = <50000>;
 		resets = <&cru SRST_TSADC>;
 		reset-names = "tsadc-apb";
-		pinctrl-names = "init", "default", "sleep";
-		pinctrl-0 = <&tsadc_otp_gpio>;
-		pinctrl-1 = <&tsadc_otp_out>;
-		pinctrl-2 = <&tsadc_otp_gpio>;
 		#thermal-sensor-cells = <1>;
 		rockchip,hw-tshut-temp = <100000>;
 		status = "disabled";
diff --git a/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10.dts b/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10.dts
index d334e21ef4ae..778cc11d99a8 100644
--- a/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10.dts
@@ -719,6 +719,9 @@
 };
 
 &tsadc {
+	pinctrl-names = "init", "default";
+	pinctrl-0 = <&tsadc_otp_gpio>;
+	pinctrl-1 = <&tsadc_otp_out>;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3326-evb-ai-va-v10.dts b/arch/arm64/boot/dts/rockchip/rk3326-evb-ai-va-v10.dts
index 3344bd61d71e..e3fb098d18d6 100644
--- a/arch/arm64/boot/dts/rockchip/rk3326-evb-ai-va-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3326-evb-ai-va-v10.dts
@@ -1140,6 +1140,9 @@
 };
 
 &tsadc {
+	pinctrl-names = "init", "default";
+	pinctrl-0 = <&tsadc_otp_gpio>;
+	pinctrl-1 = <&tsadc_otp_out>;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3326-evb-linux-lp3-v10.dts b/arch/arm64/boot/dts/rockchip/rk3326-evb-linux-lp3-v10.dts
index a1686b864fb8..2c5f6eeb52b4 100644
--- a/arch/arm64/boot/dts/rockchip/rk3326-evb-linux-lp3-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3326-evb-linux-lp3-v10.dts
@@ -837,6 +837,9 @@
 };
 
 &tsadc {
+	pinctrl-names = "init", "default";
+	pinctrl-0 = <&tsadc_otp_gpio>;
+	pinctrl-1 = <&tsadc_otp_out>;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3326-evb-lp3-v10.dts b/arch/arm64/boot/dts/rockchip/rk3326-evb-lp3-v10.dts
index 70c35052d358..636428edfce3 100644
--- a/arch/arm64/boot/dts/rockchip/rk3326-evb-lp3-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3326-evb-lp3-v10.dts
@@ -837,6 +837,9 @@
 };
 
 &tsadc {
+	pinctrl-names = "init", "default";
+	pinctrl-0 = <&tsadc_otp_gpio>;
+	pinctrl-1 = <&tsadc_otp_out>;
 	status = "okay";
 };
 
-- 
2.35.3

